src/aspire/denoising/denoiser_cov2d.py

```diff
@@ -45,9 +45,15 @@ def denoise(self, covar_opt=None, batch_size=512):
         self.cov2d = BatchedRotCov2D(self.src, self.basis, batch_size=batch_size,
                                      dtype=self.dtype)
 
-        default_opt = {'shrinker': 'frobenius_norm', 'verbose': 0, 'max_iter': 250,
-            'iter_callback': [], 'store_iterates': False, 'rel_tolerance': 1e-12,
-            'precision': 'float64'}
+        default_opt = {
+            'shrinker': 'frobenius_norm',
+            'verbose': 0,
+            'max_iter': 250,
+            'iter_callback': [],
+            'store_iterates': False,
+            'rel_tolerance': 1e-12,
```

Comment:
Sorry to come back to this again, but CG is unlikely to converge to `1e-12` in `float32`. Is that particular code path tested? It may be some issue with the CG implementation here. That code is overdue for a detailed examination.

Reply:
The `default_*_opt` are not used directly. However, the `1e-12` value is being used in the all overrides (which are just copy pastes of the defaults, lol).  I think we'd have to print diagnostic to get into what is happening there.  If it is like other scipy methods it may bail on it's own if it thinks it reached a limit (good enough for that machine precision, for example), but that is total guess.

I've created #306 for this.

Reply:
Ok thanks.